### PR TITLE
CB-16009: Nightly e2e GCP tests should use encryption

### PR DESCRIPTION
**Pull request by juhi-09**
CB-16009: Nightly e2e GCP tests should use encryption
This commit adds the e2e gcp tests for encryption key.

### CODE REVIEW

**Review by aszegedi — CHANGES_REQUESTED** [review]
Build errors happened:
```
...
> Task :integration-test:checkstyleMain
[ant:checkstyle] [ERROR] /Users/annamariaszegedi/prj/cloudbreak/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/gcp/GcpCloudFunctionality.java:9:8: Unused import - org.apache.commons.lang3.NotImplementedException. [UnusedImports]
[ant:checkstyle] [ERROR] /Users/annamariaszegedi/prj/cloudbreak/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/gcp/action/GcpClientActions.java:112:47: '{' is not preceded with whitespace. [WhitespaceAround]

> Task :integration-test:checkstyleMain FAILED
...
```
Furthermore (after I've fixed above issues) got `TestFailException`:
```
...
16:11:15.516 [TestNG-test=gcp_e2e_tests-2] ERROR c.s.i.c.t.e.d.DistroXEncryptedVolumeTest [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumesInSingleRG] - Volume has not been encrypted with 'projects/gcp-dev-cloudbreak/locations/global/keyRings/int-key/cryptoKeys/int-key' KMS key!
16:11:15.518 [TestNG-test=gcp_e2e_tests-2] ERROR c.s.i.cloudbreak.context.TestContext [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumesInSingleRG] - then [DistroXEncryptedVolumeTest$$Lambda$977/0x000000080174ef78] assertion is failed: Volume has not been encrypted with 'projects/gcp-dev-cloudbreak/locations/global/keyRings/int-key/cryptoKeys/int-key' KMS key!, name: gcp-test-f10492cb87
com.sequenceiq.it.cloudbreak.exception.TestFailException: Volume has not been encrypted with 'projects/gcp-dev-cloudbreak/locations/global/keyRings/int-key/cryptoKeys/int-key' KMS key!
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.verifyVolumeEncryptionKey(DistroXEncryptedVolumeTest.java:388)
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.verifyFreeIpaVolumeEncryptionKey(DistroXEncryptedVolumeTest.java:329)
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.lambda$testCreateDistroXWithEncryptedVolumesInSingleRG$0(DistroXEncryptedVolumeTest.java:233)
	at com.sequenceiq.it.cloudbreak.context.TestContext.then(TestContext.java:382)
	at com.sequenceiq.it.cloudbreak.dto.AbstractFreeIpaTestDto.then(AbstractFreeIpaTestDto.java:87)
	at com.sequenceiq.it.cloudbreak.dto.AbstractFreeIpaTestDto.then(AbstractFreeIpaTestDto.java:82)
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumesInSingleRG(DistroXEncryptedVolumeTest.java:233)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:238)
	at org.springframework.test.context.testng.AbstractTestNGSpringContextTests.run(AbstractTestNGSpringContextTests.java:184)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:252)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:33)
	at org.testng.internal.MethodInvocationHelper.invokeWithTimeoutWithNoExecutor(MethodInvocationHelper.java:314)
	at org.testng.internal.MethodInvocationHelper.invokeWithTimeout(MethodInvocationHelper.java:280)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:605)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
...
```
**Additional finding:** Please add the `excludedGroups: [ azure_singlerg ]` to the `resources/testsuites/e2e/gcp-e2e-tests.yaml`, like at [aws-e2e-tests.yaml](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/resources/testsuites/e2e/aws-e2e-tests.yaml#L4)

**Review by aszegedi — APPROVED** [review]

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java [hunk 1]
```diff
@@ -366,6 +380,18 @@ private void verifyVolumeEncryptionKey(CloudPlatform cloudPlatform, String resou
                     throw new TestFailException(format("Volume has not been encrypted with '%s' key!", desKeyUrl));
                 }
             });
+        } else if (CloudPlatform.GCP.equals(cloudPlatform)) {
+            String kmsKeyArn = gcpCloudProvider.getEncryptionKey(true);
```
Comment (aszegedi) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java:384]: Here you can apply a simply solution instead:
```
            String kmsKeyName = gcpCloudProvider.getEncryptionKey(true);
            List<String> volumeKmsKeyIds = new ArrayList<>(cloudFunctionality.listVolumeEncryptionKeyIds(resourceName, null, instanceIds));
            if (volumeKmsKeyIds.stream().noneMatch(keyId -> StringUtils.containsIgnoreCase(keyId, kmsKeyName))) {
                LOGGER.error(format("Volume has not been encrypted with '%s' KMS key!", kmsKeyName));
                throw new TestFailException(format("Volume has not been encrypted with '%s' KMS key!", kmsKeyName));
            } else {
                LOGGER.info(format("Volume has been encrypted with '%s' KMS key.", kmsKeyName));
                Log.then(LOGGER, format(" Volume has been encrypted with '%s' KMS key. ", kmsKeyName));
            }
```
and almost the same is true for AZURE, as a refactor:
```
            List<String> volumesDesIds = new ArrayList<>(cloudFunctionality.listVolumeEncryptionKeyIds(resourceName, resourceGroupName, instanceIds));
            if (volumesDesIds.stream().noneMatch(desId -> StringUtils.containsIgnoreCase(desId, "diskEncryptionSets/" + environmentName))) {
                LOGGER.error(format("Volume has not been encrypted with '%s' key!", desKeyUrl));
                throw new TestFailException(format("Volume has not been encrypted with '%s' key!", desKeyUrl));
            } else {
                LOGGER.info(format("Volume has been encrypted with '%s' DES key.", desKeyUrl));
                Log.then(LOGGER, format(" Volume has been encrypted with '%s' DES key. ", desKeyUrl));
            }
```

**Review by aszegedi — CHANGES_REQUESTED** [review]
Unfortunately I'm facing again an issue:
```23:59:17.510 [TestNG-test=gcp_e2e_tests-1] INFO  o.g.jersey.logging.LoggingFeature [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumes] - 1 * Sending client request on thread TestNG-test=gcp_e2e_tests-1
1 > POST http://qa-cloud-canary.eng.hortonworks.com/dl/api/sdx/gcp-test-d6dffa9603
1 > Accept: application/json
1 > Content-Type: application/json
1 > x-altus-auth: eyJhY2Nlc3Nfa2V5X2lkIjoiWTNKdU9tRnNkSFZ6T21saGJUcDFjeTEzWlhOMExURTZjV1V0WVhkek9uVnpaWEk2WTJ4dmRXUmljbVZoYXkxeFpVQm9iM0owYjI1M2IzSnJjeTVqYjIwPSIsImF1dGhfbWV0aG9kIjoiZWQyNTUxOXYxIn0=.6iRMdug7UVG2IPabTSfXXkG2oB2lwLOI8dAQM---YwWLIxcgXyaWvXgYLPj7o7op6gD7vBCgwWoHoAf5IltVDA==
1 > x-altus-date: Thu, 14 Apr 2022 21:59:17 GMT
1 > x-cdp-request-id: 3be17c19-2ee0-4ccc-bdb7-f80023f882fd
{"environment":"gcp-test-bf1d3b080ff040c4bd4","clusterShape":"LIGHT_DUTY","cloudStorage":{"adls":null,"wasb":null,"gcs":{"serviceAccountEmail":"mock-idbroker-admin-role@gcp-dev-cloudbreak.iam.gserviceaccount.com"},"s3":null,"efs":null,"adlsGen2":null,"fileSystemType":"GCS","baseLocation":"gs://cloudbreak-dev/apitest0f31bb559669492fbb68de7bcc3f1b32"},"tags":{"auto":"tesztelek","testing":"aszegedi","test-name":"testcreatedistroxwithencryptedvolumes"},"enableRangerRaz":false,"enableMultiAz":false,"runtime":"7.2.14"}
...
23:59:29.035 [TestNG-test=gcp_e2e_tests-1] INFO  o.g.jersey.logging.LoggingFeature [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumes] - 1 * Client response received on thread TestNG-test=gcp_e2e_tests-1
1 < 200
1 < cache-control: no-cache, no-store, max-age=0, must-revalidate
1 < Connection: keep-alive
1 < Content-Length: 126
1 < Content-Type: application/json
1 < Date: Thu, 14 Apr 2022 21:59:28 GMT
1 < expires: 0
1 < pragma: no-cache
1 < Server: nginx
1 < x-content-type-options: nosniff
1 < x-frame-options: DENY
1 < x-xss-protection: 1; mode=block
{"flowId":"c21ad379-156b-4143-893e-55fa9de36f69","flowChainId":null,"hasActiveFlow":true,"latestFlowFinalizedAndFailed":false}
...
23:59:39.441 [TestNG-test=gcp_e2e_tests-1] INFO  o.g.jersey.logging.LoggingFeature [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumes] - 2 * Client response received on thread TestNG-test=gcp_e2e_tests-1
2 < 200
2 < cache-control: no-cache, no-store, max-age=0, must-revalidate
2 < Connection: keep-alive
2 < Content-Length: 126
2 < Content-Type: application/json
2 < Date: Thu, 14 Apr 2022 21:59:39 GMT
2 < expires: 0
2 < pragma: no-cache
2 < Server: nginx
2 < x-content-type-options: nosniff
2 < x-frame-options: DENY
2 < x-xss-protection: 1; mode=block
{"flowId":"c21ad379-156b-4143-893e-55fa9de36f69","flowChainId":null,"hasActiveFlow":false,"latestFlowFinalizedAndFailed":true}

23:59:39.442 [TestNG-test=gcp_e2e_tests-1] ERROR c.s.it.cloudbreak.util.wait.FlowUtil [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumes] - Flow has been finalized with failed status. Crn=crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb, FlowId=c21ad379-156b-4143-893e-55fa9de36f69, FlowChainId=null
23:59:39.445 [TestNG-test=gcp_e2e_tests-1] ERROR c.s.it.cloudbreak.util.wait.FlowUtil [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumes] - Cloudbreak await for flow 'SdxTestDto[name: gcp-test-d6dffa9603]' is failed for: 'gcp-test-d6dffa9603', because of  Flow has been finalized with failed status. Crn=crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb, FlowId=c21ad379-156b-4143-893e-55fa9de36f69 , FlowChainId=null 
com.sequenceiq.it.cloudbreak.exception.TestFailException:  Flow has been finalized with failed status. Crn=crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb, FlowId=c21ad379-156b-4143-893e-55fa9de36f69 , FlowChainId=null 
	at com.sequenceiq.it.cloudbreak.util.wait.FlowUtil.waitForFlow(FlowUtil.java:125)
	at com.sequenceiq.it.cloudbreak.util.wait.FlowUtil.waitBasedOnLastKnownFlow(FlowUtil.java:70)
	at com.sequenceiq.it.cloudbreak.context.TestContext.awaitForFlow(TestContext.java:1021)
	at com.sequenceiq.it.cloudbreak.context.TestContext.awaitWithClient(TestContext.java:967)
	at com.sequenceiq.it.cloudbreak.context.TestContext.await(TestContext.java:955)
	at com.sequenceiq.it.cloudbreak.dto.sdx.SdxTestDto.await(SdxTestDto.java:141)
	at com.sequenceiq.it.cloudbreak.dto.sdx.SdxTestDto.await(SdxTestDto.java:137)
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumes(DistroXEncryptedVolumeTest.java:171)
...
```
Based on [Kibana logs](https://logs-dev-7x.sre.cloudera.com/_plugin/kibana/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-2d,to:now))&_a=(columns:!('@message'),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:manual-fields,key:context.resourceCrn,negate:!f,params:(query:'crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb'),type:phrase),query:(match_phrase:(context.resourceCrn:'crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb'))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:manual-fields,key:level,negate:!f,params:(query:error),type:phrase),query:(match_phrase:(level:error)))),index:manual-fields,interval:auto,query:(language:lucene,query:FAILED),sort:!())) the issue is
`Datalake creation failed. Database operation failed gcp-test-bf1d3b080ff040c4bd4 statusReason: Request contains an invalid argument.: [ resourceType: GCP_DATABASE,  resourceName: dbsvr-b2151156-cdc7-49ba-ab31-4c092c4252cc ]`:

```
2022-04-14 21:59:31,179 [reactorDispatcher-56] doExecute:269 ERROR c.s.d.f.c.SdxCreateActions - [type:CDPSTRUCTUREDEVENTENTITY] [crn:crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb] [name:] [flow:c21ad379-156b-4143-893e-55fa9de36f69] [requestid:eaabfa64-ff88-4211-9882-f0bc07e83cb7] [tenant:qe-aws] [userCrn:crn:altus:iam:us-west-1:qe-aws:user:cloudbreak-qe@hortonworks.com] [environment:crn:cdp:environments:eu-1:qe-aws:environment:947312c9-1c33-4817-b6f3-1cfab2bcb3af] [traceId:12cc4bd83babb8b2] [spanId:a78551639d1daaee] Datalake creation failed. Database operation failed gcp-test-bf1d3b080ff040c4bd4 statusReason: Request contains an invalid argument.: [ resourceType: GCP_DATABASE,  resourceName: dbsvr-b2151156-cdc7-49ba-ab31-4c092c4252cc ]
com.dyngr.exception.UserBreakException: Database operation failed gcp-test-bf1d3b080ff040c4bd4 statusReason: Request contains an invalid argument.: [ resourceType: GCP_DATABASE,  resourceName: dbsvr-b2151156-cdc7-49ba-ab31-4c092c4252cc ]
	at com.dyngr.core.DefaultPoller$PollerCallable.call(DefaultPoller.java:86)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at com.dyngr.concurrent.DirectExecutorService.execute(DirectExecutorService.java:55)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
	at com.dyngr.core.DefaultPoller.start(DefaultPoller.java:54)
	at com.dyngr.Polling$PollingOptions.run(Polling.java:155)
	at com.sequenceiq.datalake.service.sdx.database.DatabaseService.waitAndGetDatabase(DatabaseService.java:245)
	at com.sequenceiq.datalake.service.sdx.database.DatabaseService.waitAndGetDatabase(DatabaseService.java:237)
	at com.sequenceiq.datalake.service.sdx.database.DatabaseService.create(DatabaseService.java:133)
	at com.sequenceiq.datalake.flow.create.handler.RdsWaitHandler.doAccept(RdsWaitHandler.java:80)
	at com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler.accept(ExceptionCatcherEventHandler.java:32)
	at com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler.accept(ExceptionCatcherEventHandler.java:15)
	at com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler$$FastClassBySpringCGLIB$$d666e3b5.invoke(<generated>)
...
2022-04-14 21:59:31,159 [reactorDispatcher-96] doAccept:91 ERROR c.s.d.f.c.h.RdsWaitHandler - [type:CDPSTRUCTUREDEVENTENTITY] [crn:crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb] [name:] [flow:c21ad379-156b-4143-893e-55fa9de36f69] [requestid:eaabfa64-ff88-4211-9882-f0bc07e83cb7] [tenant:qe-aws] [userCrn:crn:altus:iam:us-west-1:qe-aws:user:cloudbreak-qe@hortonworks.com] [environment:crn:cdp:environments:eu-1:qe-aws:environment:947312c9-1c33-4817-b6f3-1cfab2bcb3af] [traceId:12cc4bd83babb8b2] [spanId:ac49e25e71bf9095] Database polling exited before timeout. Cause: 
com.dyngr.exception.UserBreakException: Database operation failed gcp-test-bf1d3b080ff040c4bd4 statusReason: Request contains an invalid argument.: [ resourceType: GCP_DATABASE,  resourceName: dbsvr-b2151156-cdc7-49ba-ab31-4c092c4252cc ]
	at com.dyngr.core.DefaultPoller$PollerCallable.call(DefaultPoller.java:86)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at com.dyngr.concurrent.DirectExecutorService.execute(DirectExecutorService.java:55)
	at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
	at com.dyngr.core.DefaultPoller.start(DefaultPoller.java:54)
	at com.dyngr.Polling$PollingOptions.run(Polling.java:155)
	at com.sequenceiq.datalake.service.sdx.database.DatabaseService.waitAndGetDatabase(DatabaseService.java:245)
	at com.sequenceiq.datalake.service.sdx.database.DatabaseService.waitAndGetDatabase(DatabaseService.java:237)
	at com.sequenceiq.datalake.service.sdx.database.DatabaseService.create(DatabaseService.java:133)
	at com.sequenceiq.datalake.flow.create.handler.RdsWaitHandler.doAccept(RdsWaitHandler.java:80)
	at com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler.accept(ExceptionCatcherEventHandler.java:32)
	at com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler.accept(ExceptionCatcherEventHandler.java:15)
	at com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler$$FastClassBySpringCGLIB$$d666e3b5.invoke(<generated>)
...
2022-04-14 21:59:31,159 [reactorDispatcher-96] lambda$waitAndGetDatabase$6:254 INFO  c.s.d.s.s.d.DatabaseService - [type:CDPSTRUCTUREDEVENTENTITY] [crn:crn:cdp:datalake:eu-1:qe-aws:datalake:072c9a06-8933-4d43-919a-2e62ce5b9dcb] [name:] [flow:c21ad379-156b-4143-893e-55fa9de36f69] [requestid:eaabfa64-ff88-4211-9882-f0bc07e83cb7] [tenant:qe-aws] [userCrn:crn:altus:iam:us-west-1:qe-aws:user:cloudbreak-qe@hortonworks.com] [environment:crn:cdp:environments:eu-1:qe-aws:environment:947312c9-1c33-4817-b6f3-1cfab2bcb3af] [traceId:12cc4bd83babb8b2] [spanId:ac49e25e71bf9095] Response from redbeams: {"environmentCrn":"crn:cdp:environments:eu-1:qe-aws:environment:947312c9-1c33-4817-b6f3-1cfab2bcb3af","name":"gcp-test-bf1d3b080ff040c4bd4-dbstck-7f09","resourceCrn":"crn:cdp:redbeams:eu-1:qe-aws:databaseServer:60c4d2c8-ce7a-4ae4-b8cc-357932bec079","status":"CREATE_FAILED","statusReason":"Request contains an invalid argument.: [ resourceType: GCP_DATABASE,  resourceName: dbsvr-b2151156-cdc7-49ba-ab31-4c092c4252cc ]"}
```

**Comment by juhi-09** [comment]
> Unfortunately I'm facing again an issue:

This is the issue with master GCP environment as well. I checked on mow-dev : https://console.dps.mow-dev.cloudera.com/cloud/environments/details/gcpwxm1/workloads

Screenshot 2022-04-19 at 1 34 30 PM

**Review by aszegedi — APPROVED** [review]
GCP encryption test have been passed for me. The proposed code change to replace the: 
```
for (int i = 0; i < volumeKmsKeyIds.size(); i++) {
                if (volumeKmsKeyIds.get(i).contains(kmsKeyArn)) {
                    LOGGER.info(format("Volume has been encrypted with '%s' KMS key.", kmsKeyArn));
                    Log.then(LOGGER, format(" Volume has been encrypted with '%s' KMS key. ", kmsKeyArn));
                } else {
                    LOGGER.error(format("Volume has not been encrypted with '%s' KMS key!", kmsKeyArn));
                    throw new TestFailException(format("Volume has not been encrypted with '%s' KMS key!", kmsKeyArn));
                }
            }
```
part with a Stream noneMatch() method was optional.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java [hunk 2]
```diff
@@ -278,6 +281,9 @@ private EnvironmentTestDto verifyEnvironmentResponseDiskEncryptionKey(TestContex
         } else if (CloudPlatform.AZURE.name().equals(environment.getCloudPlatform())) {
             String diskEncryptionSetId = environment.getAzure().getResourceEncryptionParameters().getDiskEncryptionSetId();
             verifyDiskEncryptionKey(testDto.getCloudPlatform(), diskEncryptionSetId, testDto.getRequest().getName());
+        } else if (CloudPlatform.GCP.name().equals(environment.getCloudPlatform())) {
+            String encryptionKey = environment.getGcp().getGcpResourceEncryptionParameters().getEncryptionKey();
+            verifyDiskEncryptionKey(testDto.getCloudPlatform(), encryptionKey, testDto.getRequest().getName());
```
Comment (horadla23) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java:286]: this class is full with `if` and `else if` conditions based on cloud provider. you should use `CloudFunctionality` interface instead and implement proper methods/assertions if possible for every provider

**Review by horadla23 — CHANGES_REQUESTED** [review]

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java (hunk 2 above)
Comment (juhi-09) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java]: Updated.

**Review by horadla23 — APPROVED** [review]

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java (hunk 2 above)
Comment (aszegedi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java]: Please move these to the [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java). There is already `CloudFunctionality.listVolumeEncryptionKeyIds` that is part of the already existing provider utils with the `CloudFunctionality`.

Comment (juhi-09) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java]: [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java) is used to make the cloud specific calls [CloudProvider](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java) on the other hand integrates CDP with specific cloud. These checks are CDP-per cloud specific, so I have put them in CloudProvider. There are other similar methods in AWS/Azure CloudProviders.

Please let me know if there are still concerns.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java [hunk 3]
```diff
@@ -181,4 +183,8 @@ public interface CloudProvider {
     String getFreeIpaUpgradeImageCatalog();
 
     String getStorageOptimizedInstanceType();
+
```
[the remark this answers is not in the document]
Comment (juhi-09) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java:186]: [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java) is used to make the cloud specific calls [CloudProvider](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java) on the other hand integrates CDP with specific cloud. These checks are CDP-per cloud specific, so I have put them in CloudProvider. There are other similar methods in AWS/Azure CloudProviders.

Please let me know if there are still concerns.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java (hunk 2 above)
Comment (aszegedi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java]: You were right, we cannot move the entire validation to the [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java). So I've made a refactor on these methods and fixed the `verifyDiskEncryptionKey` and `verifyVolumeEncryptionKey` calls in `CloudProviderProxy`.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (aszegedi) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java:186]: You were right, we cannot move the entire validation to the [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java). So I've made a refactor on these methods and fixed the `verifyDiskEncryptionKey` and `verifyVolumeEncryptionKey` calls in `CloudProviderProxy`.

**Review by aszegedi — APPROVED** [review]

**Pull request merged**